### PR TITLE
[FLINK-34481][table] Migrate SetOpRewriteUtil to java

### PR DESCRIPTION
**Pull request by liuyongvs**
## What is the purpose of the change

The PR migrates SetOpRewriteUtil to java for migrating 
ReplaceMinusWithAntiJoinRule
ReplaceMinusWithAntiJoinRule
RewriteIntersectAllRule
RewriteMinusAllRule

## Verifying this change

This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/SetOpRewriteUtil.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.tools.RelBuilder;
+import org.apache.calcite.util.Util;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** Util class that rewrite [[SetOp]]. */
```
Comment (JingGe) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/SetOpRewriteUtil.java:41]:
```suggestion
/** Util class that rewrite {@link org.apache.calcite.rel.core.SetOp}. */
```

Comment (liuyongvs) [reply flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/SetOpRewriteUtil.java]: done @JingGe ,thanks for your review very much

**Comment by liuyongvs** [comment]
hi @JingGe will you review it again?

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/ReplaceIntersectWithSemiJoinRule.scala [hunk 2]
```diff
@@ -49,7 +49,8 @@ class ReplaceIntersectWithSemiJoinRule
 
     val relBuilder = call.builder
     val keys = 0 until left.getRowType.getFieldCount
-    val conditions = generateEqualsCondition(relBuilder, left, right, keys)
+    val conditions =
+      generateEqualsCondition(relBuilder, left, right, keys.map(Int.box).toList.asJava)
```
Comment (JingGe) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/ReplaceIntersectWithSemiJoinRule.scala:53]: NIT: use `Integer.valueOf` instead?

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/SetOpRewriteUtil.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+        FlinkRelBuilder.pushFunctionScan(
+                relBuilder,
+                sqlFunction,
+                0,
+                relBuilder.fields(Util.range(fields.size() + 1)),
+                outputRelDataType.getFieldNames());
+
+        // correlated join
+        Set corSet = Collections.singleton(cluster.createCorrel());
```
Comment (JingGe) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/SetOpRewriteUtil.java:96]:
```suggestion
        Set<CorrelationId> corSet = Collections.singleton(cluster.createCorrel());
```

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/ReplaceIntersectWithSemiJoinRule.scala (hunk 2 above)
Comment (liuyongvs) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/ReplaceIntersectWithSemiJoinRule.scala:53]: @JingGe 
we can see two style in many scala Rules files like
    keys.map(Int.box).toList.asJava, in CommonPhysicalLookupJoin/AggregateUtil
   map(Integer.valueOf).toList.asJava in many files. 
so i change it to your suggestion

**Review by JingGe — APPROVED** [review]
LGTM

**Comment by JingGe** [comment]
@liuyongvs Thanks for your effort. Would you please to squash the commits?

**Pull request merged**